### ISSUE TITLE
02-fused-softmax: PyTorch faster than Triton on Max 1550 in composite mode 

### ISSUE THREAD

**Issue by pbchekin** [intel/intel-xpu-backend-for-triton#2363]
`ZE_FLAT_DEVICE_HIERARCHY=FLAT`
![softmax-performance](https://github.com/user-attachments/assets/a108a666-f100-4ad2-b70c-12f4e5709ab2)

`ZE_FLAT_DEVICE_HIERARCHY=COMPOSITE`
![softmax-performance](https://github.com/user-attachments/assets/dcbd5c15-1e94-4598-a934-a9936e2b45c2)

**Comment by chengjunlu** [comment]
I think it maybe caused by the register spill. 
In the composite mode, the private space for spilling maybe always allocated in one tile and all the threads on the other tile has to access the private space across the EMIB bus.

We may need to implement a new softmax kernel which is close to the torch's implementation and re-test the performance.

**Comment by anmyachev** [comment]
@pbchekin in composite mode we need to adjust cache size (double it) https://github.com/intel/intel-xpu-backend-for-triton/pull/2265#discussion_r1768397174. I wonder how the chart will change after this change.
